src/lib.rs

```diff
@@ -99,12 +108,7 @@ async fn handle(
                 .map(ToOwned::to_owned)
                 .unwrap_or_else(|| PathAndQuery::from_static(""));
 
-            let uri = Uri::builder()
-                // @TODO: Move to config or env. variable
-                .scheme("http")
-                .authority(Authority::from_static("localhost:8005"))
-                .path_and_query(path_and_query)
-                .build()?;
+            let uri = format!("{}{}", market_uri, path_and_query).parse::<Uri>()?;
```

Comment:
have you checked if `path_and_query` will add automatically `/`? maybe we can prevent errors from not passing ending `/` to the `market_uri` as we won't know if the value is ok until we actually make a request to the application.

quick solution will be to do `trim_end_matches('/')` and we will always expect the string will now have `/` (since we use it in multiple places) before we `to_string()` it in `main.rs`